doc/ipython-notebooks/multiclass/KNN.ipynb

```diff
@@ -260,14 +260,110 @@
     "knn.set_k(3)\n",
     "knn.set_knn_solver_type(KNN_COVER_TREE)\n",
     "pred = knn.apply_multiclass(feats_test)\n",
-    "print \"Covertree KNN took %2.1fs\" % (Time.get_curtime() - start)\n"
+    "print \"Covertree KNN took %2.1fs\" % (Time.get_curtime() - start)\n",
+    "\n",
+    "\n",
+    "start = Time.get_curtime()\n",
+    "knn.set_k(3)\n",
+    "knn.set_knn_solver_type(KNN_KDTREE)\n",
+    "pred = knn.apply_multiclass(feats_test)\n",
+    "print \"KDTree KNN took %2.1fs\" % (Time.get_curtime() - start)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "KNN predictions are very costly: predictiong a new point's class requires computing distances to all training samples. While Shogun exploits multi-core infrastructure, KNN quickly becomes infeasible for large datasets. We can see that the Cover Tree is faster than standard KNN. In contrast, the KD-Trees solver is much slower. This is likely to be caused be the high-dimensional data that we used. We now use low-dimensional data to verify it."
```

Comment:
We can see that the Cover Tree is faster than standard KNN --- should be
We can see that the cover tree solver is faster than standard KNN


This is likely to be caused be the high-dimensional data that we used. --- should be
This is likely to be caused by the high-dimensional data that we used.